core/base/src/TApplication.cxx

```diff
@@ -628,21 +630,416 @@ TApplication::EExitOnException TApplication::ExitOnException(TApplication::EExit
+            // We need to append "constexpr"!
+            if ((func->Property() & kIsConstexpr)>0) {
+               md5Text.Prepend("constexpr ");
+            }
+         }
+         md5Text.Append(" ");
+      }
+   md5Text.Append(className);
+   md5Text.Append("::");
+   md5Text.Append(methodName);
+   md5Text.Append(methodName);
+   // We use "FormatMethodForDoxygen" to modify the arguments of Method with respect of Doxygen.
+   md5Text.Append(FormatMethodForDoxygen(className, func));
+   // We generate the URL for the class/namespace.
+   className.ReplaceAll("::","_1_1");
+   TString url = UrlGenerator(className, scopeType);
+   url.Append("#a");
+   url.Append(md5Text.MD5());
+   return url;
+}
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// The function lists useful commands (".help") or opens the online reference
+/// guide, generated with Doxygen, for a certain class (".help className"),
+/// method of a class (".help className::Method") or a namespace
+/// (".help NameSpace").
+///
+/// \param[in] line command from the command line
+
+void TApplication::Help(const char *line)
+{
+   // We first check if the user wants to print the help on the interpreter.
+   TString command(line);
+   TString strippedCommand = command.Strip(TString::kBoth);
+   // If the user chooses ".help" or ".?".
+   if ((strippedCommand == ".help") || (strippedCommand == ".?")) {
+      gInterpreter->ProcessLine(line);
+   //   " << metaString << "help ClassName\t\t- Opens Doxygen online reference guide for ClassName\n"
+   //   "   " << metaString << "help ClassName::Method\t- Opens Doxygen online reference guide for Method\n"
+   //   "   " << metaString << "help NameSpace\t\t- Opens Doxygen online reference guide for NameSpace\n"
+      Printf("\nROOT special commands.");
+      Printf("==========================================================================");
+      Printf("   pwd                 : show current directory, pad and style");
+      Printf("   ls                  : list contents of current directory");
+      Printf("   which [file]        : shows path of macro file");
+      Printf("   .help Class         : opens the reference gude for that class");
+      Printf("   .help Class::Member : opens the reference gude for function/member");
+      return;
+   }
+
+   // If the user wants to use the extended ".help className" command to access
+   // the online reference guide.
+   if (!strippedCommand.BeginsWith(".help ")) {
+      if (!strippedCommand.BeginsWith(".? ")) {
+         Error("Help", "Unknown command!");
+         return;
+      }
+   }
+   // We remove the command ".help" or ".?" from the TString.
+   if (strippedCommand.BeginsWith(".? ")){
+      strippedCommand.Remove(0, 3);
+   } else {
+      strippedCommand.Remove(0, 5);
+   }
+   // We strip the TString and substract the name of the method.
+   TString strippedClass = strippedCommand.Strip(TString::kBoth);
```

Comment:
You should be able to do `command = command.Strip(TString::kBoth);` instead of introducing many more variables.